### PR TITLE
Centralize init

### PR DESCRIPTION
**Pull request by mgermain**

### CODE REVIEW

File: platoon/channel.py [hunk 1]
```diff
@@ -49,26 +35,38 @@ class Controller(object):
...
+        ## Cleanup and init global lock and job_uid name ##
+        self._job_uid = "platoon_{0}_{1}".format(os.path.basename(os.path.expanduser('~')), control_port)
+
+        # The ExistentialError is apparently the only way to verify if the semaphore/shared_memory exists.
+        try:
+            posix_ipc.unlink_semaphore(self._job_uid+"lock")
+        except posix_ipc.ExistentialError:
+            pass
+        # Initializing lock
+        posix_ipc.Semaphore(self._job_uid+"lock", posix_ipc.O_CREAT, initial_value=1)
```
Comment (abergeron) [inline platoon/channel.py]: No, I disagree with that.  This should be in the worker, not the controller.

File: platoon/channel.py [hunk 2]
```diff
@@ -248,20 +259,34 @@ def init_control_sock(self, port):
...
+        _ffi = cffi.FFI()
+        _ffi.cdef("void *mmap(void *, size_t, int, int, int, size_t);")
+        _lib = _ffi.dlopen(None)
+
+        addr = _ffi.NULL
+
+        m = _lib.mmap(addr, length, prot, flags, fd, offset)
+        if m == _ffi.cast('void *', -1):
+            raise OSError(_ffi.errno, "for mmap")
+        return _ffi.buffer(m, length)
```
Comment (abergeron) [inline platoon/channel.py]: Defining a mapping with cffi is not free so this is why I do it at module level.  Now you re-do it every this this is called.

It may not be so bad since it won't be called often, but why did you change that?

File: platoon/channel.py (hunk 1 above)
Comment (mgermain) [inline platoon/channel.py]: Why?

File: platoon/channel.py [hunk 3]
```diff
@@ -248,20 +259,34 @@ def init_control_sock(self, port):
...
-        Intialize shared memory parameters.
+        Initialize shared memory parameters.
 
         This must be called before accessing the params attribute
         and/or calling :meth:`sync_params`, :meth:`lock_params` or
         :meth:`unlock_params`.
 
-        Paramters
+        Parameters
         ---------
```
Comment (abergeron) [inline platoon/channel.py]: You have to extend the underline too for proper format.

File: platoon/channel.py (hunk 2 above)
Comment (mgermain) [inline platoon/channel.py]: It's called once per worker, if you really think it's too much I'll put is back in the module and add some comments that explain the reason.

File: platoon/channel.py (hunk 3 above)
Comment (mgermain) [inline platoon/channel.py]: Oops thanks.

File: platoon/channel.py (hunk 1 above)
Comment (nouiz) [inline platoon/channel.py]: This change was done to allow starting all worker at the same time. I don't see problems with that and not doing this would make thing more complicated elsewhere I think.

File: platoon/channel.py (hunk 2 above)
Comment (mgermain) [inline platoon/channel.py]: Wait, this was called every time the module was imported so it was executed once per worker anyway.
No?

Comment (abergeron) [inline platoon/channel.py]: Yes, but now it may be called more than once.

File: platoon/channel.py (hunk 1 above)
Comment (abergeron) [inline platoon/channel.py]: I guess this would work since everything is on the same machine.  But it doesn't feel like a proper separation.

File: example/lstm.py [hunk 4]
```diff
@@ -506,18 +505,15 @@ def train_lstm(
...
 
     # This creates Theano Shared Variable from the parameters.
     # Dict name (string) -> Theano Tensor Shared Variable
     # params and tparams have different copy of the weights.
     tparams = init_tparams(params)
 
-    worker.init_shared_params('DLTlstm', tparams.values(),
-                              param_sync_rule=EASGD(0.5),
-                              cleanup=init)
+    worker.init_shared_params(tparams.values(), param_sync_rule=ASGD())
```
Comment (abergeron) [inline example/lstm.py]: Why did you switch the example update rule?

Comment (nouiz) [inline example/lstm.py]: Because we aren't able to get any training speed up with EASGD now. So we should more recommand to use ASGD

Comment (mgermain) [inline example/lstm.py]: Simply because @carriepl email was saying that we should use this one.

Comment (abergeron) [inline example/lstm.py]: But it works in the example :)

But ok.

Comment (mgermain) [inline example/lstm.py]: I just did some more testing and I'll switch back to EASGD for now.
